module/src/hooks/calendar/calendar.utils.ts

```diff
@@ -98,30 +105,18 @@ export function getMonths(forYear: number, min?: Date, max?: Date, locale: Local
  * Returns a list of years to display within the year select of the calendar view.
  * @param min (Date) The optional `min` date prop passed to the calendar view (parsed to a `Date` object from a `DateLike`). If no min passed, will default to `1900`
  * @param max (Date) The optional `max` date prop passed to the calendar view (parsed to a `Date` object from a `DateLike`). If no min passed, will default to `2100`
- * @returns An array of years as numbers.
+ * @returns An array of year objects containing all necessary display characteristics and statuses.
  */
-export function getYears(min?: Date, max?: Date): number[] {
-  const minYear = min?.getFullYear() ?? 1900;
-  const maxYear = max?.getFullYear() ?? 2100;
-  const years: number[] = [];
-  for (let i = minYear; i <= maxYear; i += 1) {
-    years.push(i);
+export function getYears(min?: Date, max?: Date): IYear[] {
+  minMaxCheckThrow(min, max);
+  let currentYear = min ? startOfYear(min) : new Date(1900, 1, 1);
+  const years: IYear[] = [];
+  while (currentYear.getFullYear() <= (max?.getFullYear() ?? 2100)) {
+    years.push({
+      date: currentYear,
+      number: currentYear.getFullYear(),
+    });
+    currentYear = addYears(currentYear, 1);
```

Comment:
could this be done with a repeat? Like

```ts
Array.repeat(startOfYear(max).getFullYear() ?? 1900 - startOfYear(min).getFullYear() ?? 2100, index => {
   ...return get year from index
})
```

Reply:
The first two lines will crash if `min` and `max` are undefined, and the repeat is non-inclusive so will leave years out that needs to be in.

What I've done works and is perfectly readable, I see no reason to change it.